src/api/auth/token.py

```diff
@@ -0,0 +1,25 @@
+import logging
+
+from fastapi import HTTPException, Request, status
+
+from src.settings import settings
+
+
+async def check_token(request: Request):
+    if not hasattr(settings, "ACCESS_TOKEN_FOR_PROCHARITY"):
+        return
+    match request.headers.get("token"):
+        case None:
+            status_code, log_message, message = (
+                status.HTTP_401_UNAUTHORIZED,
+                "Request without any token provided",
+                "Request without token",
+            )
```

Comment:
Будет лучше читаться если мы в кейсах логику напишем, а не ниже. 
```suggestion
             await logging.ainfo("Request without any token provided")
             raise HTTPException(
                status_code=status.HTTP_401_UNAUTHORIZED,
                detail="Request without token",
            )
```